molgenis-app/src/main/java/org/molgenis/app/promise/model/PromiseMappingProjectMetaData.java

```diff
@@ -0,0 +1,232 @@
+package org.molgenis.app.promise.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.molgenis.MolgenisFieldTypes;
+import org.molgenis.app.promise.mapper.PromiseMapperFactory;
+import org.molgenis.data.AggregateQuery;
+import org.molgenis.data.AggregateResult;
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.MolgenisDataException;
+import org.molgenis.data.Query;
+import org.molgenis.data.Repository;
+import org.molgenis.data.RepositoryCapability;
+import org.molgenis.data.RepositoryDecorator;
+import org.molgenis.data.support.DefaultEntityMetaData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PromiseMappingProjectMetaData extends DefaultEntityMetaData
+{
+	public static final String ENTITY_NAME = "PromiseMappingProject";
+	public static final String FULLY_QUALIFIED_NAME = PromisePackage.NAME + '_' + ENTITY_NAME;
+
+	public static final String NAME = "name";
+	public static final String BIOBANK_ID = "biobank_id";
+	public static final String CREDENTIALS = "Credentials";
+	public static final String MAPPER = "mapper";
+
+	public static final String LBL_NAME = "Name";
+	public static final String LBL_BIOBANK_ID = "Biobank ID";
+	public static final String LBL_CREDENTIALS = "Credentials";
+	public static final String LBL_MAPPER = "Mapper";
+
+	@Autowired
+	public PromiseMappingProjectMetaData(PromiseMapperFactory promiseMapperFactory)
+	{
+		super(ENTITY_NAME, PromisePackage.getPackage());
+
+		requireNonNull(promiseMapperFactory);
+
+		setLabel("ProMISe mapping projects");
+		setDescription("");
+
+		addAttribute(NAME).setIdAttribute(true).setNillable(false).setLabel(LBL_NAME)
+				.setDescription("The name of this mapping");
+		addAttribute(BIOBANK_ID).setNillable(false).setLabel(LBL_BIOBANK_ID)
+				.setDescription("The ID of the biobank in the BBMRI-NL Sample Collections entity").setUnique(true);
+		addAttribute(CREDENTIALS).setDataType(MolgenisFieldTypes.XREF).setRefEntity(PromiseCredentialsMetaData.INSTANCE)
+				.setNillable(false).setLabel(LBL_CREDENTIALS)
+				.setDescription("The ProMISe credentials for this biobank");
+		addAttribute(MAPPER).setNillable(false).setLabel(LBL_MAPPER)
+				.setDescription("The mapper to use for this biobank");
+
+		setDecorator(new RepositoryDecorator()
```

Comment:
We have so many decorators, we really need an DefaultRepositoryDecorator that simply delegates everything. Then other decorators can extend this and only change the behavior that needs changing.